git-config/src/file/git_config.rs

```diff
@@ -1673,6 +1747,191 @@ mod from_paths {
+            a_path.as_path(),
+            "
+            [core]
+              a = false
+              sslVerify = true
+              d = 41",
+        )
+        .unwrap();
+
+        let b_path = dir.path().join("b");
+        let relative_b_path = std::path::PathBuf::from("b");
+        fs::write(
+            b_path.as_path(),
+            "
+            [diff]
+              renames = true",
+        )
+        .unwrap();
+
+        let a_path_string = a_path
+            .as_path()
+            .parent()
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .replace("\\", "\\\\");
+        let non_canonical_path_a = format!("{}/./a", a_path_string);
+        let non_existing_path = "/dfgwfsghfdsfs";
+        let c_path = dir.path().join("c");
+        fs::write(
+            c_path.as_path(),
+            format!(
+                "
+            [core]
+              c = 12
+              d = 42
+            [include]
+              path = {}
+              path = {}
+              path = {}
+            [http]
+	          sslVerify = false",
+                non_existing_path,
+                non_canonical_path_a,
+                relative_b_path.as_path().to_str().unwrap()
+            ),
+        )
+        .unwrap();
+
+        let config = GitConfig::from_paths(&[c_path.as_path()], None).unwrap();
+
+        assert_eq!(
+            config.get_raw_value("core", None, "c"),
+            Ok(Cow::<[u8]>::Borrowed(b"12"))
+        );
+        assert_eq!(
+            config.get_raw_value("core", None, "d"),
+            Ok(Cow::<[u8]>::Borrowed(b"41"))
+        );
+        assert_eq!(
+            config.get_raw_value("http", None, "sslVerify"),
+            Ok(Cow::<[u8]>::Borrowed(b"false"))
+        );
+
+        assert_eq!(
+            config.get_raw_value("diff", None, "renames"),
+            Ok(Cow::<[u8]>::Borrowed(b"true"))
+        );
+
+        assert_eq!(
+            config.get_raw_value("core", None, "a"),
+            Ok(Cow::<[u8]>::Borrowed(b"false"))
+        );
+    }
+
+    #[test]
+    fn nested_include_has_max_depth_10() {
+        let dir = tempdir().unwrap();
+        let path = dir.path().join("0");
+        fs::write(
+            path.as_path(),
+            format!(
+                "
+                [core]
+                  i = true 
+                [include]
+                  path = {}",
+                path.as_path().to_str().unwrap().replace("\\", "\\\\")
+            ),
+        )
+        .unwrap();
+
+        for i in 1..11 {
+            let path = dir.path().join(&i.to_string());
+            let prev_path = dir.path().join(&(i - 1).to_string());
+            fs::write(
+                path.as_path(),
+                format!(
+                    "
+                    [core]
+                      i = false 
+                    [include]
+                      path = {}",
+                    prev_path.as_path().to_str().unwrap().replace("\\", "\\\\")
```

Comment:
Since this conversion is done a lot, I think it would be better to have a little function for it that is used to describe why we do that. I'd think it's just important for windows and it could say so either with a descriptive function name or a short function name and some description for the function itself.